src/common.rs

```diff
@@ -981,9 +981,45 @@ pub fn get_current_pid() -> Result<Pid, &'static str> {
     inner()
 }
 
+/// MAC address for network interface
+#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+pub struct MacAddr {
+    data: [u8; 6],
+}
+
+impl MacAddr {
```

Comment:
One thought I just had: we could maybe add a helper method `is_unspecified` which would basically do `self == Self::UNSPECIFIED`. What do you think?